articles/public-key-cryptography-in-js/index.md

```diff
@@ -0,0 +1,223 @@
+Public key cryptography, also known as `asymmetric cryptography` is an encryption system comprising of public and private key pairs. The private keys are known to the sender and receiver of information only whereas the public keys are disseminated widely. It solves the [key distribution problem](https://www.open.edu/openlearn/ocw/mod/oucontent/view.php?id=48322&section=1.3) brought about by symmetric cryptography.
+
+#### Prerequisites
+- Have [Node.js](https://nodejs.org/en/) installed on your computer.
+- Have some basic knowledge of JavaScript programming language.
+- Have some basic knowledge of cryptography.
+
+### What we will cover
+- [Introduction](#introduction)
+- [Algorithms using public key cryptography](#algorithms-using-public-key-cryptography)
+- [Libraries for public key cryptography in JavaScript](#libraries-for-public-key-cryptography-in-javascript)
+- [Using TweetNaCl.js to implement public key cryptography](#using-tweetnacl.js-to-implement-public-key-cryptography)
+- [Man-in-the-middle attack](#man-in-the-middle-attack)
+- [Using pre-computed keys](#using-pre-computed-keys)
+- [Maintaining Public keys](#maintaining-public-keys)
+- [Maintaining Private or Secret keys](#maintaining-private-or-secret-keys)
+- [Platforms that employ public key cryptography](#platforms-that-employ-public-key-cryptography)
+
+### Introduction
+Cryptography refers to the encoding and decoding of messages so that to maintain confidentiality, integrity, and authentication of information in transit. Public key cryptography is one of the methods of cryptography. In this method, there is the public and private key. The public key is known widely whereas the private key is secret to a communicating pair. When a pair wants to communicate, the sender encrypts the message with the public key of the recipient to come up with the ciphertext. When the recipient receives the message, he or she decrypts the message using his or her private key. The key is not transported individually, it is encoded in the ciphertext. This eliminates the [Key distribution problem](https://www.open.edu/openlearn/ocw/mod/oucontent/view.php?id=48322&section=1.3)
```

Comment:
`The key is not transported individually, it is encoded in the ciphertext.`

In this line, can you be more specific on which key is encoded in the ciphertext? The public or the private key.

Reply:
resolved